packages/openapi-sampler/tests/VendorExamples.spec.ts

```diff
@@ -0,0 +1,252 @@
+import { VendorExtensions } from '../src/VendorExtensions';
+import { VendorExamples } from '../src/traverse';
+
+describe('VendorExamples', () => {
+  describe('find', () => {
+    let sut!: VendorExamples;
+
+    beforeEach(() => {
+      sut = new VendorExamples();
+    });
+
+    it.each([
+      {
+        [VendorExtensions.X_EXAMPLE]: {
+          name: 'name',
+          age: 30
+        }
+      },
+      {
+        [VendorExtensions.X_EXAMPLE]: {
+          'application/json': {
+            name: 'name',
+            age: 30
+          }
+        }
+      },
+      {
+        [VendorExtensions.X_EXAMPLES]: {
+          'application/json': {
+            name: 'name',
+            age: 30
+          }
+        }
+      },
+      {
+        [VendorExtensions.X_EXAMPLES]: {
+          'application/json': {
+            some: {
+              name: 'name',
+              age: 30
+            }
+          }
+        }
+      },
+      {
+        [VendorExtensions.X_EXAMPLES]: {
+          'application/json': {
+            some: {
+              name: 'name',
+              age: 30
+            }
+          }
+        }
+      },
+      {
+        [VendorExtensions.X_EXAMPLES]: [
+          {
+            name: 'name',
+            age: 30
+          }
+        ]
+      },
+      {
+        [VendorExtensions.X_EXAMPLES]: {
+          'application/json': [
+            {
+              name: 'name',
+              age: 30
+            }
+          ]
+        }
+      }
+    ])('should match %j object vendor example', (input) => {
+      // arrange
+      const expected = {
+        name: 'name',
+        age: 30
+      };
+
+      const schema = {
+        type: 'object',
+        properties: {
+          name: {
+            type: 'string'
+          },
+          age: {
+            type: 'integer'
+          }
+        },
+        ...input
+      };
+
+      // act
+      const result = sut.find(schema);
+
+      // assert
+      expect(result).toMatchObject(expected);
```

Comment:
```suggestion
      expect(result).toEqual(expected);
```
as we do not expect extra properties